server/api/schema.ts

```diff
@@ -106,6 +108,64 @@ const typeDefs = gql`
     #     NOT_AVAILABLE: Boolean
     # }
 
+    enum DonationItemCondition {
+        BRAND_NEW
+        GREAT
+        GOOD
+        FAIR
+        POOR
+    }
+
+    enum DonationItemStatus {
+        PENDING_APPROVAL
+        PENDING_DROPOFF
+        PENDING_MATCH
+    }
+
+    type DonationFormContact {
+        firstName: String
+        lastName: String
+        email: String
+        phoneNum: String
+    }
+
+    type DonationForm {
+        _id: ID
+        contact: DonationFormContact
+        name: String
+        requestGroup: RequestGroup
+        description: String
+        quantity: Int
+        age: Int
+        condition: DonationItemCondition
+        images: [String]
+
+        adminNotes: String
+        status: DonationItemStatus
+        quantityMatched: Int
```

Comment:
Based on the designs for donation form matching, `quantityRemaining` instead of `quantityMatched` might be better. We only display the quantity remaining
Thoughts

Reply:
yup that sounds good! do we need to keep a `quantityRemaining` field on the `Request` model to keep track of how much of the request still needs to be matched, or would it be better to calculate that through the information in the `matchedDonations`?

similarly where would we put the logic for setting the `request.fulfilledAt` field when the request is fully matched? I'm thinking that it could be done on the backend when we match a donation form to a request, but it could also be updated on the frontend - not sure which is best

Reply:
sorry for the late response!

>  do we need to keep a quantityRemaining field on the Request model to keep track of how much of the request still needs to be matched, or would it be better to calculate that through the information in the matchedDonations?

We'll calculate the quantity remaining for a request from `matchedDonations`

> similarly where would we put the logic for setting the request.fulfilledAt field when the request is fully matched?

Hmmm let's do it in the frontend. there's a `fulfillRequest` endpoint that we can call after the `matchedDonations` field on a request is updated

We are trying to make the backend a little dumber. If we put the logic in the backend, then we would have to detect when `matchedDonations` on a request is changed in the `updateRequest` resolver and then change `fulfilledAt` if necessary